From: Ajit Khaparde <ajit.khaparde@emulex.com>
Date: Thu, 26 Apr 2012 15:42:39 +0000
Subject: [PATCH 064/143] be2net: fix speed displayed by ethtool on certain
 SKUs
Patch-mainline: v3.5-rc1
Git-commit: 2a89611a051cdcfcf4a7ae21d18eef64fab517b4
References: bnc#777565 FATE#313819

logical speed returned by link_status_query needs to be multiplied by 10.

Signed-off-by: Ajit Khaparde <ajit.khaparde@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/emulex/benet/be_ethtool.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/emulex/benet/be_ethtool.c
+++ b/drivers/net/ethernet/emulex/benet/be_ethtool.c
@@ -557,7 +557,7 @@ static int be_get_settings(struct net_de
 			if (!status)
 				be_link_status_update(adapter, link_status);
 			if (link_speed)
-				et_speed = link_speed;
+				et_speed = link_speed * 10;
 			else
 				et_speed = convert_to_et_speed(port_speed);
 		} else {
